src/igvfd/audit/file.py

```diff
@@ -112,3 +112,34 @@ def audit_file_no_file_format_specifications(value, system):
             f'has no `file_format_specifications`.'
         )
         yield AuditFailure(audit_message.get('audit_category', ''), f'{detail} {audit_message.get("audit_description", "")}', level=audit_message.get('audit_level', ''))
+
+
+@audit_checker('IndexFile', frame='object')
+@audit_checker('AlignmentFile', frame='object')
+@audit_checker('TabularFile', frame='object')
+@audit_checker('ReferenceFile', frame='object')
+@audit_checker('ModelFile', frame='object')
+@audit_checker('SequenceFile', frame='object')
+def audit_file_missing_anvil_url(value, system):
+    '''
+    [
+        {
+            "audit_description": "Released and archived files that are restricted access are expected to be loaded in anvil.",
+            "audit_category": "missing anvil url",
+            "audit_level": "INTERNAL_ACTION"
+        }
+    ]
+    '''
+    object_type = space_in_words(value['@type'][0]).capitalize()
+    audit_message = get_audit_message(audit_file_missing_anvil_url, index=0)
+    if value.get('status') in ['released', 'archived'] and value.get('controlled_access', False) and not (value.get('anvil_url')):
+        file_set_object = system.get('request').embed(
+            value.get('file_set'), '@@object_with_select_calculated_properties?field=data_use_limitation_summaries')
+        if file_set_object.get('status') in ['released', 'archived']:
+            data_use_limitation_summaries = file_set_object.get('data_use_limitation_summaries', '')
+            if data_use_limitation_summaries and data_use_limitation_summaries != ['no certificate']:
+                detail = (
+                    f'{object_type} {audit_link(path_to_text(value["@id"]), value["@id"])} '
```

Comment:
Think would be nice to have clear indication of which Anvil project a file needs to be moved to based on its DUL code or data use summary etc. But maybe that's still being worked out.

Reply:
I can add it to the detail but for your usage I think the facet will be more useful. The detail is just displayed in the audit

Reply:
I agree that the detail will be nice but the facet will be more useful